File: benchmarks/fixtures/js_project/tests/strings.test.js
import { test } from 'node:test';
import assert from 'node:assert/strict';
import { reverse, countWords, capitalize, camelCase } from '../src/strings.js';

test('reverse("hello") === "olleh"', () => { assert.equal(reverse('hello'), 'olleh'); });
test('countWords("hello world") === 2', () => { assert.equal(countWords('hello world'), 2); });
test('capitalize("hello") === "Hello"', () => { assert.equal(capitalize('hello'), 'Hello'); });
test('camelCase("hello world") === "helloWorld"', () => {
    assert.equal(camelCase('hello world'), 'helloWorld');
});
test('camelCase("foo bar baz") === "fooBarBaz"', () => {
    assert.equal(camelCase('foo bar baz'), 'fooBarBaz');
});

File: benchmarks/fixtures/go_project/stringutils/strings_test.go
package stringutils

import "testing"

func TestReverse(t *testing.T) {
	if got := Reverse("hello"); got != "olleh" {
		t.Errorf("Reverse(%q) = %q, want %q", "hello", got, "olleh")
	}
}

func TestCountWords(t *testing.T) {
	if got := CountWords("hello world foo"); got != 3 {
		t.Errorf("CountWords(%q) = %d, want 3", "hello world foo", got)
	}
}

func TestTitleCase(t *testing.T) {
	cases := [][2]string{
		{"hello world", "Hello World"},
		{"foo bar baz", "Foo Bar Baz"},
		{"go is great", "Go Is Great"},
	}
	for _, c := range cases {
		got := TitleCase(c[0])
		if got != c[1] {
			t.Errorf("TitleCase(%q) = %q, want %q", c[0], got, c[1])
		}
	}
}
